File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
import static java.lang.Math.sqrt;

class Solution {
    public long solution(long n) {
        double s = sqrt(n);
        if (s == (long) s) {
            long next = (long) s + 1;
            return next * next;
        } else {
            return -1;
        }
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.kt
import kotlin.math.sqrt

class Solution {
    fun solution(n: Long): Long {
        var s = sqrt(n.toDouble())
         return if (s == s.toLong().toDouble()) {
            (s.toLong() + 1) * (s.toLong() + 1)
        } else {
             -1
        }
    }
}